### ISSUE TITLE
Wrong yaml schema for group_vars?

### ISSUE THREAD

**Issue by tsia** [ansible-community/vscode-ansible#36]
i have opened a Folder containing a playbook in vscode.
inside a subfolder named `group_vars` i have `all.yml` containing some variables.

vscode-ansible complains about the format because it uses the "Ansible Playbook Schema" for some reason:
![image](https://user-images.githubusercontent.com/1386595/109825552-38edfb80-7c3a-11eb-9b7a-f5cbbad09044.png)

i saw you have set `yamlValidation` accordingly in your package.json so i don't know what is happening here.

**Comment by ssbarnea** [comment]
I have no idea why it gets confused as based on https://github.com/ansible-community/vscode-ansible/blob/master/package.json#L104-L115 it should pick the first pattern that matches, which is groups_vars. Do you happen to have these under a path that looks like `playbooks/group_vars/all.yml` ? -- is if you do, it would still have to work.

**Comment by tsia** [comment]
The `group_vars` folder actually is at the root of the project.

**Comment by ssbarnea** [comment]
And the full past of disk? can you try to create few more file-paths and get more info? I hope we did not encountered a bug in vscode itself.

**Comment by ssbarnea** [comment]
I can confirm the bug, in fact it seems that whatever I do vars files seems to endup using the playbook schema. Looking at the config, I see no reason for this to happen.

The bug seems related to a recent merge made to https://github.com/redhat-developer/yaml-language-server/pull/417 and I wonder if next release of YAML extension will adress it,... or make things even worse.

Test cases:
* `playbooks/all.yml` - should use ansible-playbook schema 
* `group_vars/all.yml` - should use ansible-vars schema
* `playbooks/group_vars/all.yml` - should use ansible-vars schema, even if both patterns are matching. I do think there is a real chance this may go wrong as I do remember that vscode original globbing was not very smart and a single `*` could expand to anything. I am also not sure if the order of the patterns is respected either. We do list playbooks at the end for good reasons: to avoid folders like vars/tasks/handlers nested inside playbooks/ to endup being recognized as playbooks.


I hope that @JPinkney or @joshuawilson may give us some insights here. Maybe the can help us by adding some extra tests that assure we can write patterns to distinguish between those cases.

**Comment by JPinkney** [comment]
I can probably take a look as to what's causing on the yaml side later today

**Comment by ssbarnea** [comment]
I tested with vscode-ansible@master and I can confirm that now case 1 and 2 are working, but 3 is not. `playbooks/group_vars/all.yml` is still identified as a playbook instead of a vars file and I do not know how to define patterns that avoid overlapping.

If you look at https://github.com/ansible-community/vscode-ansible/blob/master/package.json#L117 you will see that vars pattern is listed before the playbook one, so it should be picked first, right?

**Comment by JPinkney** [comment]
> If you look at https://github.com/ansible-community/vscode-ansible/blob/master/package.json#L117 you will see that vars pattern is listed before the playbook one, so it should be picked first, right?

It evaluates everything equally, there is no way to prioritise one or the other. Let me try it with https://github.com/redhat-developer/yaml-language-server/pull/417 and i'll see if that fixes it

**Comment by JPinkney** [comment]
I played around with `playbooks/group_vars/all.yml` and it looks like both schemas (vars and playbook) are currently being set. You can see this if you do:
```
allUsers
  aa
```

you'll see an error with:
```
Ansible Vars Schema

Incorrect type. Expected "ansible-vars.json".yaml-schema: Ansible Vars Schema
Incorrect type. Expected "array".yaml-schema: Ansible Playbook Schema
```

**Comment by bendem** [comment]
I have a problem which I think might be related, if not I'll open another issue:
![image](https://user-images.githubusercontent.com/2681677/111149258-f35ff580-858c-11eb-9e4e-ed266f37fc66.png)
This is for a yaml file in `ansible/playbooks/sso/tasks/openldap/`.
I have the same problem with vars of all types (group_vars, host_vars, vars, defaults):
![image](https://user-images.githubusercontent.com/2681677/111149550-4df95180-858d-11eb-9b77-a81f67204626.png)

Basically, anything under playbooks/* is interpreted as a playbook, no matter the parent directory.

On that note, it might be cool if things under files and templates were left alone since they are not actually ansible files.

**Comment by EpiqSty** [comment]
It looks like `vars_prompt` also affected:
with ` a most basic example:` from https://docs.ansible.com/ansible/latest/user_guide/playbooks_prompts.html 
```
---
- hosts: all
  vars_prompt:

    - name: username
      prompt: What is your username?
      private: no

    - name: password
      prompt: What is your password?

  tasks:

    - name: Print a message
      ansible.builtin.debug:
        msg: 'Logging in as {{ username }}'
```

I'm getting this error:

![image](https://user-images.githubusercontent.com/25151779/115889849-9b9e8f00-a454-11eb-8659-d48585a8e626.png)

```
$ apt-cache policy code
code:
  Installed: 1.55.2-1618307277
```
redhat.vscode-yaml Preview v0.18.0
zbr.vscode-ansible Preview 0.3.2


Please let me know if it's not related and needs a separated issue.

**Comment by ssbarnea** [comment]
> It looks like `vars_prompt` also affected:
> with ` a most basic example:` from https://docs.ansible.com/ansible/latest/user_guide/playbooks_prompts.html
> 
> ```
> ---
> - hosts: all
>   vars_prompt:
> 
>     - name: username
>       prompt: What is your username?
>       private: no
> 
>     - name: password
>       prompt: What is your password?
> 
>   tasks:
> 
>     - name: Print a message
>       ansible.builtin.debug:
>         msg: 'Logging in as {{ username }}'
> ```
> 
> I'm getting this error:
> 
> ![image](https://user-images.githubusercontent.com/25151779/115889849-9b9e8f00-a454-11eb-8659-d48585a8e626.png)
> 
> ```
> $ apt-cache policy code
> code:
>   Installed: 1.55.2-1618307277
> ```
> 
> redhat.vscode-yaml Preview v0.18.0
> zbr.vscode-ansible Preview 0.3.2
> 
> Please let me know if it's not related and needs a separated issue.

The bug is valid and the correct place to raise it is at https://github.com/ansible-community/schemas -- as this is missing from the schema. It should not be hard to add it.
